documentation/mqtt-overview.md

```diff
@@ -0,0 +1,83 @@
+
+## M5 Specifics
+
+### MQTT Deployment
+
+AnkerMake has deployed MQTT via [EMQX](https://www.emqx.com/en). Most likely it's [EMQX Cloud](https://www.emqx.com/en/cloud) since the servers are hosted by [AWS](https://aws.amazon.com/). We got that info via typing the following command into a terminal:
+
+```
+nslookup make-mqtt.ankermake.com
+```
+
+That returns something similar to:
+
+```
+Non-authoritative answer:
+Name:    ankermake-emqx-nlb-60dd062d3b52769a.elb.us-east-2.amazonaws.com
+Addresses:  3.22.247.85
+          3.134.119.140
+          18.189.81.10
+Aliases:  make-mqtt.ankermake.com
+```
+
+As you can see in the `Name:` field, it's an EMQX instance running on an AWS server.
+
+### Update Behavior
+
+As long as a M5 is powered on, it publishes updates to the broker servers under the `/phone/maker/YOUR_SERIAL_NUMBER_HERE/notice` topic as a client device. The update seems to be done by exception, meaning that it only updates when a value changes.
+
+The M5 also subscribes to  commands that may be sent to it from the MQTT servers for execution.
+
+Another client device (your phone or computer) subscribes to topics associated with the target printer and can publish commands for execution on the target printer.
+
+The basic relationships for MQTT communications can be illustrated as such:
+
+```mermaid
+graph
+    A("[Broker] Ankermake MQTT Servers (Job Queue and Data)") <-- "MQTT Data (TLS)" --> B(("M5 Printers"))
+    A <-- "MQTT Data (TLS)" --> C(("Client Devices"))
+    D("AnkerMake HTTPS Servers (API)") -- "MQTT Login Info" --> C
+    C -- "Authentication Token" --> D
+```
+
+We need to utilize the AnkerMake HTTPS API to pull down the MQTT login info and MQTT encryption key needed to setup a new client device connection instance. Your **Authentication Token** (auth_token) is what you need to authenticate your identity with the AnkerMake HTTPS servers. Then, we can plug in those MQTT credentials into the MQTT server to subscribe/publish to the topics in the broker.
+
+The messages are exchanged via an encrypted tunnel (standard MQTT over TLS) but part of the payload is further encrypted using standard AES-256-CBC using a MQTT only encryption key.
+
+### Payload Structure
+
+In the `specifications` folder, you'll find a file named `mqtt.stf` and that's what we're using to map out the contents of the message payload. We don't know a whole lot right now about the possibilities of what you could see in these payloads, but we've managed to map some of the standard scenarios out. 
+
+The entire payload is delivered as a series of raw bytes and we attempt to unwrap that into something useful. You can do whatever post processing to this packet you like to separate it into useful parts. In our example file, we pretty much just label the entire packet based on our best guess.
```

Comment:
I don't understand this paragraph.

Are we talking about the general case, where mqtt can contain anything? That's true, but we only have a few unknown bytes left in the message. After parsing and decryption, we always (in every sample I've seen) get perfectly well-formed JSON messages out as a result.

What example file are you talking about?

(It's true that PPPP so far requires a bit more guesswork, but that's unrelated to the mqtt situation)

Reply:
Gotcha, I think I was getting confused about how much we know about the messages with PPPP. I'll clean this up